### PR TITLE
[stable/kong] add Kong Enterprise support

### PR DESCRIPTION
**Pull request by rainest**
Signed-off-by: Travis Raines <traines@konghq.com>

#### What this PR does / why we need it:

Adds support for Kong Enterprise and toggling the chart between a Kong open source deployment (default) or Kong Enterprise deployment. 

#### Special notes for your reviewer:
@hbagdi and @shashiranjan84 for review. This PR is pre-squashed; https://github.com/rainest/charts/tree/kong-enterprise has additional history and https://github.com/Kong/charts/pull/1 has some prior discussion.

These additions are designed to support Kong Enterprise 0.35+. 

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/chart]`)

### CODE REVIEW

File: stable/kong/Chart.yaml [hunk 1]
```diff
@@ -10,5 +10,5 @@ maintainers:
 name: kong
 sources:
 - https://github.com/Kong/kong
-version: 0.11.2
+version: 0.12.2
```
Comment (hbagdi) [inline stable/kong/Chart.yaml]: Please bump to 0.12.0

Comment (rainest) [reply stable/kong/Chart.yaml]: Fixed in https://github.com/helm/charts/pull/14509/commits/041d75feb0cd119d4ba7a0bb2dfe689288e1c381

**Comment by hbagdi** [comment]
Thank you @rainest for this big addition!

**Pull request merged**